presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayElementBlockWriter.java

```diff
@@ -139,6 +139,12 @@ public void reset(BlockBuilderStatus blockBuilderStatus)
     }
 
     @Override
+    public BlockBuilder resetNew(BlockBuilderStatus blockBuilderStatus)
+    {
+        return new ArrayElementBlockWriter(blockBuilder.resetNew(blockBuilderStatus), start);
```

Comment:
I'm pretty sure this should not be implemented and instead should `throw new UnsupportedOperationException()`